File: python/ch01/fibonacci/fibonacci/fibonacci_5.py

def fibonacci(n: int) -> int:
  if  n == 0: return n

  last: int = 0
  next: int = 1

  for _ in range(1, n):
    last, next = next, last + next 
  
  return next

def run():
  print("***** Algorithm: Iterative Calculation *****")

  print(f"fibonacci(0) = {fibonacci(0)}")
  print(f"fibonacci(1) = {fibonacci(1)}")
  print(f"fibonacci(5) = {fibonacci(5)}")
  print(f"fibonacci(20) = {fibonacci(20)}")
  print(f"fibonacci(50) = {fibonacci(50)}")

File: rust/ch01/fibonacci/src/fibonacci_5.rs
pub fn fibonacci(n: usize) -> usize {
  if n == 0 {
    return n;
  }

  let mut last: usize = 0;
  let mut next: usize = 1;

  for _ in 1..n {
    let temp: usize = last;
    last = next;
    next = temp + next;
  }
  next
}

pub fn run() {
  println!("***** Algorithm: Iterative Calculation *****");

  println!("fibonacci(0) = {}", fibonacci(0));
  println!("fibonacci(1) = {}", fibonacci(1));
  println!("fibonacci(5) = {}", fibonacci(5));
  println!("fibonacci(20) = {}", fibonacci(20));
  println!("fibonacci(50) = {}", fibonacci(50));
}
